### DISCUSSION TITLE
Merge Maps Cache and Download

### DISCUSSION THREAD

**Discussion by alensiljak** [organicmaps/organicmaps#2278] [Ideas]
Could someone be so kind as to explain why the following is not practical? (Assuming it **is** impractical since it has not been implemented)

As has been mentioned in #446, there are practical reasons to use online maps. One does not need to download the whole region, it provides a quick search in areas that are not available offline.
Additional layer of goodness is caching the previewed areas offline. That way, the area we've looked at is available later when we might not have an internet connection available.

An interesting combination is when whole regions are downloaded for offline use.

I am wondering why not store maps in the same format, irrespective of whether they have been explicitly downloaded or just cached by the app. If the maps were in the same format, we could even update certain areas that were previously downloaded, without the full download of the updated map. Or before the new package is published.
It would be possible to have small pockets of maps that we need, without downloading whole countries or regions.

What would be some practical pros and cons of this approach?

I reckon that data compression has to do with it. But isn't gzipped HTTP compression already dealing with that part?

**Comment by biodranik** [comment]
To display a map you need to download it first. Most maps take 25-250 Mb of space. Frustrated users will uninstall the app for such a slow "online" experience.

**Comment by alensiljak** [comment]
Magic Earth does the same and I haven't uninstalled it yet.😉

Haven't tried explicitly downloading the maps yet, though, to see if they are already using this concept.

**Reply by alensiljak** [reply]
StreetComplete is even more interesting as it pulls updates quicker. I noticed that, when I'm adding info in another tool, SC gets the new data on the next refresh.
